pilot/pkg/proxy/envoy/v1/mixer.go

```diff
@@ -262,21 +303,33 @@ func mixerHTTPRouteConfig(mesh *meshconfig.MeshConfig, role model.Node, instance
 }
 
 // Mixer TCP filter config for inbound requests.
-func mixerTCPConfig(role model.Node, check bool, instance *model.ServiceInstance) *FilterMixerConfig {
+func buildTCPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Node, instance *model.ServiceInstance) *FilterMixerConfig {
 	filter := &FilterMixerConfig{
 		MixerAttributes: map[string]string{
 			AttrDestinationIP:  role.IPAddress,
 			AttrDestinationUID: "kubernetes://" + role.ID,
 		},
 	}
+
+	// TODO (Bug): DisablePolicyChecks is being ignored in this function
+	transport := &mccpb.TransportConfig{
+		CheckCluster: PolicyCheckClusterName,
+		ReportCluster: TelemetryClusterName,
+	}
+	if mesh.PolicyCheckServer == mesh.TelemetryServer {
+		transport.ReportCluster = transport.CheckCluster
+	}
```

Comment:
Looks like this is added for backwards compat. in which case a comment to that effect would be helpful.

Reply:
don't understand.. what backward compat? I thought transport was part of the mixer filter's v2 config

Reply:
I was trying to understand why `transport.ReportCluster` is overwritten if the Check/Report server addresses are the same.